sycl/source/detail/host_pipe_map_entry.hpp

```diff
@@ -0,0 +1,46 @@
+//==----------------- host_pipe_map_entry.hpp --------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <cstdint>
+#include <unordered_map>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace detail {
+
+// Forward declaration
+class device_impl;
+
+struct HostPipeMapEntry {
+  // Pointer to the host_pipe on host.
+  void *MHostPipePtr;
+  // Size of the underlying type in the host_pipe.
+  std::uint32_t MHostPipeTSize;
+  // True if the host_pipe has been decorated with device_image_scope
+  bool MIsDeviceImageScopeDecorated;
+
+  // Constructor only initializes with the pointer to the host_pipe as the
+  // additional information is loaded after.
+  HostPipeMapEntry(void *HostPipePtr)
+      : MHostPipePtr(HostPipePtr), MHostPipeTSize(0),
+        MIsDeviceImageScopeDecorated(false) {}
+
+  void initialize(std::uint32_t HostPipeTSize,
+                  bool IsDeviceImageScopeDecorated) {
+    assert(HostPipeTSize != 0 && "Host pipe initialized with 0 size.");
+    assert(MHostPipeTSize == 0 && "Host pipe has already been initialized.");
+    MHostPipeTSize = HostPipeTSize;
+    MIsDeviceImageScopeDecorated = IsDeviceImageScopeDecorated;
+  }
+};

```

Comment:
Main place of interest: Is there other information that needed to be encoded in each pipe_id to pipe_pointer mapping. Currently it only keeps track of the underlying data type's size and whether is device image scope decorated (which I doubt if it is needed).